### PR TITLE
bug: where: check `condition` is boolean

### PR DESCRIPTION
**Pull request by lucascolley**
Towards https://github.com/data-apis/array-api-extra/issues/49, https://github.com/scipy/scipy/pull/21783

Blocked on https://github.com/data-apis/array-api/pull/868

cc @ev-br @asmeurer

### CODE REVIEW

**Comment by lucascolley** [comment]
some of the failures may be alleviated by existing changes on array-api-tests' main branch

File: array_api_strict/_searching_functions.py [hunk 1]
```diff
@@ -80,6 +80,9 @@ def where(condition: Array, x1: Array, x2: Array, /) -> Array:
     """
     # Call result type here just to raise on disallowed type combinations
     _result_type(x1.dtype, x2.dtype)
+    
+    if condition.dtype != bool:
```
Comment (ev-br) [inline array_api_strict/_searching_functions.py:84]: Should be `array_api_strict.bool != builtins.bool`

**Review by ev-br — CHANGES_REQUESTED** [review]
Bizzare that `where` has no smoke tests here. gh-78 adds some.

File: array_api_strict/_searching_functions.py (hunk 1 above)
Comment (lucascolley) [reply array_api_strict/_searching_functions.py:84]: Ah good catch, I missed the override of the built-in

**Comment by ev-br** [comment]
Thanks Lucas!

**Pull request merged**